src/ir/auth_logic/universe_relation_insertion.cc

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2022 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "src/ir/auth_logic/universe_relation_insertion.h"
+
+#include "src/ir/auth_logic/is_name_constant.h"
+
+namespace raksha::ir::auth_logic {
+
+namespace {
+
+std::string TypeToUniverseName(datalog::ArgumentType arg_type) {
```

Comment:
`const datalog::ArgumentType arg_type&` (We are not making a copy of arg_type here.)

Reply:
fixed